### PR TITLE
Get route

### PR DESCRIPTION
**Pull request by DanielGluskin**
- getById route
- test /post using /get to verify the new order is retrievable
- send email to admin on successful new order
- test /post to verify email was send to admin on successful order

### CODE REVIEW

File: example-application/test/basic-tests.test.js [hunk 1]
```diff
@@ -120,7 +142,25 @@ describe("/api", () => {
...
+      const orderToAdd = {
+        userId: 1,
+        mode: "draft",
+      };
+
+      //Act
+      const orderAddResult = await request(expressApp).post("/order").send(orderToAdd);
+
+      //Assert
+      expect(orderAddResult.status).toBe(400);
```
Comment (goldbergyoni) [inline example-application/test/basic-tests.test.js:160]: We don't stand to the test title promise - We only test the response. Shouldn't we check that it's not retrievable?

**Review by goldbergyoni — CHANGES_REQUESTED** [review]
@DanielGluskin Overall great PR, left few comments, some are minor, some maybe worth a discussion. Curious to hear your thoughts

File: example-application/test/basic-tests.test.js [hunk 2]
```diff
@@ -72,29 +91,33 @@ describe("/api", () => {
       });
 
       //Act
-      const receivedAPIResponse = await request(expressApp).post("/order").send(orderToAdd);
+      const addedOrderId = (await request(expressApp).post("/order").send(orderToAdd)).body.id;
+      const retreivedAddedOrder = await request(expressApp).get("/order/" + addedOrderId);
```
Comment (goldbergyoni) [inline example-application/test/basic-tests.test.js:95]: @DanielGluskin This is part of the assert phase. Act phases contains always 1 call against the method/route under test

File: example-application/test/basic-tests.test.js [hunk 3]
```diff
@@ -26,40 +26,59 @@ beforeEach(() => {
...
   // ️️️✅ Best Practice: Clean-up resources after each run
   await stopWebServer();
+  nock.enableNetConnect();
   done();
 });
 
 // ️️️✅ Best Practice: Structure tests
 describe("/api", () => {
+
+  describe("GET /orders", () => {
```
Comment (goldbergyoni) [inline example-application/test/basic-tests.test.js:44]: @DanielGluskin There are no GET tests

File: example-application/test/basic-tests.test.js [hunk 4]
```diff
@@ -26,40 +26,59 @@ beforeEach(() => {
 });
 
 afterEach(() => {
+  process.env.SEND_MAILS = "false";
```
[the remark this answers is not in the document]
Comment (DanielGluskin) [reply example-application/test/basic-tests.test.js:29]: We had another test doing:
```
//Arrange
process.env.SEND_MAILS = "true";
```
which was never cleaned (or restored to default). What if the default is indeed FALSE?
I totaly agree with the point we cannot assume what is the production value. Perhaps we should capture the default value inside the beforeAll, and restore it after every test?

```
beforeAll(() => { defaultSendMails = process.env.SEND_MAILS
afterEach(() => { process.env.SEND_MAILS = defaultSendMails
```

File: example-application/test/basic-tests.test.js (hunk 1 above)
Comment (DanielGluskin) [reply example-application/test/basic-tests.test.js:160]: How do we check it if we don't receive any id from the post request?

File: example-application/test/basic-tests.test.js [hunk 5]
```diff
@@ -72,29 +91,33 @@ describe("/api", () => {
...
-      const receivedAPIResponse = await request(expressApp).post("/order").send(orderToAdd);
+      const addedOrderId = (await request(expressApp).post("/order").send(orderToAdd)).body.id;
+      const retreivedAddedOrder = await request(expressApp).get("/order/" + addedOrderId);
 
       //Assert
-      const { status, body } = receivedAPIResponse;
+      const { body } = retreivedAddedOrder;
 
       expect({
-        status,
```
[the remark this answers is not in the document]
Comment (DanielGluskin) [reply example-application/test/basic-tests.test.js:81]: Do you mean we also should check the status of the get response? Shouldn't we test it inside the get test?

File: example-application/test/basic-tests.test.js (hunk 3 above)
Comment (DanielGluskin) [reply example-application/test/basic-tests.test.js]: Added.

File: example-application/test/basic-tests.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (goldbergyoni) [reply example-application/test/basic-tests.test.js:29]: Great point. Maybe something generic that captures the entire process.env before and restores.

Let's open an issue and take it outside of this PR? For now, I would do something super-simple or nothing at all

File: example-application/test/basic-tests.test.js [hunk 6]
```diff
@@ -26,75 +26,117 @@ beforeEach(() => {
...
+        userId: 1,
+        productId: 2,
+        mode: "approved",
+      };
+
+      //Act
+      const { body: { id: addedOrderId } } = await request(expressApp).post("/order").send(orderToAdd);
+
+      //Assert
+      const { body, status } = await request(expressApp).get("/order/" + addedOrderId);
```
Comment (goldbergyoni) [inline example-application/test/basic-tests.test.js:120]: Use template literals:

`/order/${addedOrderId}`

File: example-application/test/basic-tests.test.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (goldbergyoni) [reply example-application/test/basic-tests.test.js:81]: @DanielGluskin Oh yes, no reason to check the status of a call that is just meant for assertion. You're right

File: example-application/test/basic-tests.test.js (hunk 1 above)
Comment (goldbergyoni) [reply example-application/test/basic-tests.test.js:175]: There is usually a way to retrieve a record, some field can be set with unique value. For example, IF order had a field called order.remarks we could do order.remarks = something-unique, query for it and ensure nothing returns. 

In our case, we have too few fields, I would consider enriching the payload a bit, like adding "remarks", "totalToPay"

p.s. we didn't fill the field "productId"

**Review by goldbergyoni — APPROVED** [review]
@DanielGluskin We could discuss these things for days, many interesting points, I would for now fix the easier parts and merge. For others, just open an issue.

**Pull request merged**